tasks/gulp/__tests__/check-compatibility-with-govuk-frontend-version.test.js

```diff
@@ -1,21 +1,35 @@
 /* eslint-env jest */
-const { writeFileSync, unlink } = require('fs');
+const { writeFileSync, promises: { rm, mkdtemp } } = require('fs');
 const path = require('path');
 const { exec } = require('child_process');
-
-const initCwd = path.join(process.env.INIT_CWD, 'tasks/gulp/__tests__');
+const { tmpdir } = require('os');
 
 describe('govuk-frontend version compatibility check', () => {
-  const scriptPath = path.resolve(__dirname, '../../../check-compatibility-with-govuk-frontend-version.js');
-  const mockPackagePath = path.resolve(__dirname, 'package.json');
-  const createMockPackage = (contents) => writeFileSync(mockPackagePath, JSON.stringify(contents));
+  const compatibilityCheckScript = path.resolve(__dirname, '../../../check-compatibility-with-govuk-frontend-version.js');
+
+  let tmpDir;
 
-  afterAll(() => {
-    unlink(mockPackagePath, () => {});
+  beforeEach(async () => {
+    tmpDir = await mkdtemp(path.join(tmpdir(), 'hmrc-frontend-unit-tests-'));
   });
 
-  beforeAll(() => {
-    process.env.HMRC_FRONTEND_DISABLE_COMPATIBILITY_CHECK = '';
```

Comment:
basically you could just delete this beforeAll I think

Reply:
just me unhelpfully backseat driving with that comment from PR 🙅

Reply:
but hopefully useful example in any case of tmpdir making stuff! though we only need to use tmpdirs since we're running subprocess, where most tests would mock the fs I think rather than creating actual files